dlog/commitment/src/commitment.rs

```diff
@@ -75,9 +77,22 @@ where
     }
 }
 
-pub fn shift_scalar<F: PrimeField>(x: F) -> F {
-    let two: F = 2_u64.into();
-    x - two.pow(&[F::Params::MODULUS_BITS as u64])
+pub fn shift_scalar<G: AffineCurve>(x: G::ScalarField) -> G::ScalarField
+where
+    G::BaseField: PrimeField,
+{
+    let n1 = <G::ScalarField as PrimeField>::Params::MODULUS;
+    let n2 = <G::ScalarField as PrimeField>::BigInt::from_bits_le(
+        &<G::BaseField as PrimeField>::Params::MODULUS.to_bits_le()[..],
+    );
+    let two: G::ScalarField = (2 as u64).into();
+    if n1 < n2 {
+        (x - (two.pow(&[<G::ScalarField as PrimeField>::Params::MODULUS_BITS as u64])
+            + G::ScalarField::one()))
```

Comment:
from https://github.com/MinaProtocol/mina/blob/39dd86401bb56ee06a6004fe8ac4f952050d4a6d/src/lib/pickles_types/shifted_value.ml#L34 my understanding is that it should be `- G::ScalarField::one()` not `+`

Reply:
The parentheses handle negation, this is correct as written.

Reply:
oh, damn parenthesis :)